src/validator.js

```diff
@@ -231,6 +235,18 @@ export default class Validator {
         })
         .then(() => {
           this.xpi = new _Xpi(this.packagePath);
+          return this.xpi.getMetaData();
+        })
+        .then((metadata) => {
+          var chromeManifest = 'chrome.manifest';
+          if (Object.keys(metadata).indexOf(chromeManifest) > -1) {
+            return this.scanFile(chromeManifest, 'stream');
+          } else {
+            // TODO: Log this.
```

Comment:
I think we'll need to work this out for the layout tests following actually knowing what extension type we're validating it. Tracking down precisely what the rules are for this might be a pain. But we should probably use the existing validator a guide.